pkg/storage/segments/repairer.go

```diff
@@ -160,27 +164,37 @@ func (repairer *Repairer) Repair(ctx context.Context, path storj.Path) (err erro
 			NodeId:   node.Id,
 			Hash:     hashes[i],
 		})
+		healthyMap[int32(i)] = true
 	}
 
-	// Update the remote pieces in the pointer
-	pointer.GetRemote().RemotePieces = healthyPieces
-
-	length := int32(len(healthyPieces))
+	healthyLength := int32(len(healthyPieces))
 	switch {
-	case length <= pointer.Remote.Redundancy.RepairThreshold:
+	case healthyLength <= pointer.Remote.Redundancy.RepairThreshold:
 		mon.Meter("repair_failed").Mark(1)
-	case length < pointer.Remote.Redundancy.SuccessThreshold:
+	case healthyLength < pointer.Remote.Redundancy.SuccessThreshold:
 		mon.Meter("repair_partial").Mark(1)
 	default:
 		mon.Meter("repair_success").Mark(1)
 	}
 
 	healthyRatioAfterRepair := 0.0
 	if pointer.Remote.Redundancy.Total != 0 {
-		healthyRatioAfterRepair = float64(len(healthyPieces)) / float64(pointer.Remote.Redundancy.Total)
+		healthyRatioAfterRepair = float64(healthyLength) / float64(pointer.Remote.Redundancy.Total)
 	}
 	mon.FloatVal("healthy_ratio_after_repair").Observe(healthyRatioAfterRepair)
 
+	// if partial repair, include "unhealthy" pieces that are not duplicates
+	if healthyLength < pointer.Remote.Redundancy.SuccessThreshold {
+		for _, p := range unhealthyPieces {
```

Comment:
Can we prioritize offline nodes over disqualified nodes? If not easy to do right now, let's at least add a TODO comment.